code_generation/data/attribute_classes/update.json

```diff
@@ -147,6 +147,21 @@
             "name": "FaultUpdate",
             "base": "BaseUpdate",
             "attributes": [
+                {
+                    "data_type": "IntS",
+                    "names": "status",
+                    "description": "Whether the fault is connected"
+                },
+                {
+                    "data_type": "FaultType",
+                    "names": "fault_type",
+                    "description": "type of the fault"
+                },
+                {
+                    "data_type": "FaultPhase",
+                    "names": "fault_phase",
+                    "description": "phase(s) of the fault"
+                },
```

Comment:
should these both be part of the `update`?

Reply:
My feeling is that a fault is a certain type of fault, so you should not be able to change the fault type and phase, only the `status`. So if you want to test various faults on 1 location you should specify multiple faults and switch them on or off. 
But this is a design choice. What do you think?
Also curious what @TonyXiang8787 thinks about this

Reply:
i'm on your boat for the time being, but i wonder what that means for - e.g. - choosing the fault phase. i feel like if we make `fault_phase` updateable, then we should do the same for the `fault_type`